pkg/autodiscovery/autoconfig.go

```diff
@@ -563,13 +559,30 @@ func (ac *AutoConfig) resolveTemplateForService(tpl integration.Config, svc list
 	return resolvedConfig, nil
 }
 
-// GetLoadedConfigs returns configs loaded
-func (ac *AutoConfig) GetLoadedConfigs() map[string]integration.Config {
+// WithLoadedConfigs calls the given function with the map of all
+// loaded configs.  This is done with the config store locked, so
+// callers should perform minimal work within f.
+func (ac *AutoConfig) WithLoadedConfigs(f func(map[string]integration.Config)) {
```

Comment:
I don't find the name very helpful. What do you think of `MapOnLoadedConfigs` ?

Reply:
It is patterned after pkg/collector/runner/tracker/tracker.go:WithCheck and WithRunningChecks.  How about MapOver..?